src/Doctrine/Schema/DcaSchemaProvider.php

```diff
@@ -190,6 +206,11 @@ private function parseColumnSql(Table $table, string $columnName, string $sql):
             $collation = $match[1];
         }
 
+        // Use the binary collation if the BINARY flag is set (see #1286)
+        if (preg_match('/\bbinary\b/i', $def, $match)) {
```

Comment:
Should this be supported for non-VARCHAR fields? Also, if a collation is already set, I assume it should not be overridden?

Reply:
> Should this be supported for non-VARCHAR fields?

Yes, of course. It should be supported for every field that supports collations (usually all text fields).

> Also, if a collation is already set, I assume it should not be overridden?

I don't think so. Having both `COLLATE` and `BINARY` is a configuration error.

Reply:
But then we should invert the if's, so that `COLLATE` overrides `BINARY` imho.

Reply:
No. If the `BINARY` flag is set, it means "use the binary collation". Doesn't matter what collation is given IMHO.

But why don't we just throw an exception? It is a configuration error.

Reply:
Changed in d8f4653d71baba848b2b4b598dd2dc7d5a341e14.

Reply:
Nope, we should not throw an error message but do the same as MySQL does if we execute the following query:

```sql
ALTER TABLE `tl_article` ADD `test` VARCHAR(255) BINARY COLLATE utf8mb4_unicode_ci NOT NULL;
```

This is how the field is created:

<img width="425" alt="" src="https://user-images.githubusercontent.com/1192057/34833492-d346d91e-f6ef-11e7-972a-19bc7f302dde.png">

The `binary` flag has precedence over the `collate` function, so the initial code was correct.